src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,478 @@
+
+  https://github.com/babashka/nbb/blob/bca8b5017a06768eb35d02a2d6233ca9c6c2f692/src/nbb/impl/nrepl_server.cljs"
+  (:require [basilisp.contrib.bencode :as bc]
+            [basilisp.string :as str])
+  (:import [basilisp.__version__ :as ver]
+           [basilisp.lang.runtime :as rt]
+           socketserver
+           sys
+           traceback
+           uuid))
+
+(def ^:private log-levels
+  "A compile time set used to determine which log levels should be
+  included in the output."
+  #{:warn})
+
+;; helper compile time macros to log ``values`` according to the level
+;; as filtered by `log-levels`.
+(defmacro ^:private debug [& values]
+  (when (some #{:debug} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "debug") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private info [& values]
+  (when (some #{:info} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "info") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private warn [& values]
+  (when (some #{:warn} log-levels)
+    `(.write sys/stdout (str ~(keyword (str *ns*) "warn") " " (str/join " " [~@values]) \newline))))
+(defmacro ^:private error [& values]
+  ;; we want error to always output to `sys/stderr` unfiltered.
+  `(.write sys/stderr (str ~(keyword (str *ns*) "error") " " (str/join " " [~@values]) \newline)))
+
+(definterface IReadWrite
+  ;; Pythonic interface for creating `sys/stdout` like File objects.
+  (read [])
+  (write [value]))
+
+(deftype StreamOutFn [out-fn]
+  ;; A type to use as replacement binding for writing to `sys/stdout`
+  ;; stream, so that the output ``value`` is passed to ``out-fn``
+  ;; instead.
+  IReadWrite
+  (read [_this]
+    (throw (python/RuntimeError (str ::logic-error " shouldn't be possible to get here..."))))
+  (write [_self value]
+    (out-fn value)))
+
+(defn- response-for-mw [handler]
+  (fn [{:keys [id session] :as request} response]
+    (let [response (cond-> (assoc response
+                                  "id" id)
+                     session (assoc "session" session))]
+      (handler request response))))
+
+(defn- coerce-request-mw [handler]
+  (fn [request send-fn]
+    (handler (update request :op keyword) send-fn)))
+
+(defn- log-request-mw [handler]
+  (fn [request send-fn]
+    (info :request (dissoc request :client*))
+    (handler request send-fn)))
+
+(defn- log-response-mw [handler]
+  (fn [request response]
+    (info :response response)
+    (handler request response)))
+
+(declare ops)
+
+(defn- handle-describe [request send-fn]
+  (send-fn request
+           {"versions" {"basilisp" (let [version (py->lisp ver/VERSION)]
+                                     (assoc (zipmap ["major" "minor" "incremental"]
+                                                    version)
+                                            "version-string" (str/join "." version)))
+                        "python" (let [version (.__getitem__ (.split sys/version " ") 0)]
```

Comment:
Does the CLJ `get` (backed by [this runtime function](https://github.com/basilisp-lang/basilisp/blob/ebfa81ba8a2b237242d044ec4f09b1835f2fb1fa/src/basilisp/lang/runtime.py#L1173-L1201)) not work here? I am noticing we don't handle `tuple`s in the singledispatch, so that could be why. I don't think we should have to fall back to `__getitem__` in most cases.

Reply:
Actually it works with `get` and updated as such, I think it was a remnant of my previous attempt to get a version number.